src/components/forms/DatePicker/DatePicker.stories.tsx

```diff
@@ -0,0 +1,114 @@
+
+Because this component uses the useEffect hook to trigger validation whenever the date value changes (regardless of how), the React DatePicker will validate when:
+- setting the initial value based on the default value passed in (same as above)
+- clicking on a date in the calendar UI (same as above)
+- on input (change) of the external text input
+
+It's also worth mentioning that validation in this case is just calling [setCustomValidity](https://developer.mozilla.org/en-US/docs/Web/API/HTMLObjectElement/setCustomValidity) on the external text input, and library users should be able to determine how & when they want invalid UI to display by inspecting the [ValidityState](https://developer.mozilla.org/en-US/docs/Web/API/ValidityState) of the external input.
+
+We may find that we want to expose props for custom event handlers or even a ref to the component for better integration with 3rd party form libraries. If you are running into this, please [file an issue](https://github.com/trussworks/react-uswds/issues/new/choose) describing your use case.
+`,
+      },
+    },
+  },
+}
+
+export const completeDatePicker = (argTypes): React.ReactElement => (
+  <Form onSubmit={argTypes.onSubmit}>
+    <FormGroup>
+      <Label id="appointment-date-label" htmlFor="appointment-date">
+        Appointment date
+      </Label>
+      <div className="usa-hint" id="appointment-date-hint">
+        mm/dd/yyyy
+      </div>
+      <DatePicker
+        id="appointment-date"
+        name="appointment-date"
+        aria-describedby="appointment-date-label appointment-date-hint"
+        disabled={argTypes.disabled}
+      />
+    </FormGroup>
+  </Form>
+)
+
+export const defaultDatePicker = (): React.ReactElement => (
+  <DatePicker id="birthdate" name="birthdate" />
+)
+
+export const disabled = (): React.ReactElement => (
+  <DatePicker id="birthdate" name="birthdate" disabled />
+)
+
+export const withDefaultValue = (): React.ReactElement => (
+  <DatePicker id="birthdate" name="birthdate" defaultValue="1988-05-16" />
+)
+
+export const withDefaultInvalidValue = (): React.ReactElement => (
+  <DatePicker
+    id="birthdate"
+    name="birthdate"
+    defaultValue="1988-05-16"
+    minDate="2020-01-01"
+  />
+)
+
+export const withMinMaxInSameMonth = (): React.ReactElement => (
+  <DatePicker
+    id="birthdate"
+    name="birthdate"
+    minDate="2021-01-10"
+    maxDate="2021-01-20"
+  />
+)
+
+export const withMinMax = (): React.ReactElement => (
+  <DatePicker
+    id="birthdate"
+    name="birthdate"
+    minDate="2020-01-01"
+    maxDate="2021-5-31"
+  />
+)
+
+export const withRangeDate = (): React.ReactElement => (
+  <DatePicker
+    id="birthdate"
+    name="birthdate"
+    defaultValue="2021-01-20"
+    rangeDate="2021-01-08"
+  />
+)
+
+export const withExternalInput = (): React.ReactElement => (
+  <>
+    <DatePicker id="birthdate" name="birthdate" />
+    <input type="text" data-testid="test-external-element" />
+  </>
```

Comment:
@brandonlenz I _think_ this was to demonstrate that the Calendar closes if focus moves to another element and/or opening the Calendar creates a focus trap (so you can't tab to the next input until it's closed). I probably just added it to test and don't mind removing if it's confusing

Reply:
Ohhh! Got it!

That makes sense. 

I'd say it's a valuable story to have. I would consider renaming the story to be more specific to how it relates to focusing.

I was confused because I was expecting the external input to be magically hooked into the value/validation of the `DatePicker` itself (since `DatePicker` internally has an "externalInput") for some use case in this story 🤦

Reply:
I got rid of this story and added a second input to the "Complete Datepicker" example to hopefully illustrate the same thing in a less confusing way!